shared/request.ts

```diff
@@ -16,30 +16,41 @@ export async function request(
   method: 'GET' | 'POST' | 'PATCH',
   path: `/${string}`,
   body?: Record<string, unknown>,
+  timeout?: undefined | number,
```

Comment:
Let's add timeout as last parameter as it's optional.

Reply:
After this change, I am hoping QR login will not work as we are sending ESIGNET_HOST as last argument in function call. Please check it once.

Reply:
I'm a bit confused with **the definitive/recommended order** of parameters in typescript and I've not found the guidelines on the order of parameters of different types such as `normal`, `optional` and `default`.